### PR TITLE
[release-4.14] USHIFT-1779: fix golang check in spec file

### PR DESCRIPTION
**Pull request by dhellmann**
Limit the golang version check to the major and minor version numbers.

/assign @ggiguash @pacevedom @pmtk

### CODE REVIEW

File: packaging/rpm/microshift.spec [hunk 1]
```diff
@@ -127,8 +127,8 @@ The microshift-greenboot package provides the Greenboot scripts used for verifyi
 
 %prep
 # Dynamic detection of the available golang version also works for non-RPM golang packages
-golang_detected=$(go version | awk '{print $3}' | tr -d '[a-z]')
-golang_required=%{golang_version}
+golang_detected=$(go version | awk '{print $3}' | tr -d '[a-z]' | cut -f1-2 -d.)
+golang_required=$(echo %{golang_version} | cut -f1-2 -d.)
```
[the remark this answers is not in the document]
Comment (dhellmann) [reply packaging/rpm/microshift.spec:131]: The output of go version includes all 3 parts so we have to strip both. 

Could we just remove the whole thing? Why do we need this check?

**Review by pacevedom — APPROVED** [review]
/lgtm
/label backport-risk-assessed
/label cherry-pick-approved

**Comment by dhellmann** [comment]
Re-trying the labels since it seems the bot had a glitch a little while ago

/label backport-risk-assessed
/label jira/valid-bug

**Comment by dhellmann** [comment]
I used staff engineer privileges to add the cherry-pick-approved and lgtm labels. @pacevedom had previously commented to add those but the bot missed the notice. We need this fix to unblock changes on the main branch.

**Pull request merged**